File: app/src/main/java/com/wealoha/social/beans/common/ConstantsService.java
package com.wealoha.social.beans.common;

import retrofit.Callback;
import retrofit.http.GET;

import com.wealoha.social.beans.Result;
import com.wealoha.social.impl.ServerUrlImpl;

/**
 * 
 * 
 * @author javamonk
 * @see
 * @since
 * @date 2014-11-20 下午5:51:11
 */
public interface ConstantsService {

	@GET(ServerUrlImpl.CONSTANTS)
	public void get(Callback<Result<ConstantsData>> callback);

	/**
	 * API可用的入口
	 * 
	 * @return
	 */
	@GET(ServerUrlImpl.CONSTANTS_API_ENDPOINT)
	public Result<ApiEndpointData> apiEndpoing();

}

File: app/src/main/java/com/wealoha/social/api/ConstantsService.kt
package com.wealoha.social.api

import com.wealoha.social.beans.Result
import com.wealoha.social.beans.common.ApiEndpointData
import com.wealoha.social.beans.common.ConstantsData
import com.wealoha.social.impl.ServerUrlImpl
import retrofit.Callback
import retrofit.http.GET

/**
 *
 *
 * @author javamonk
 * @see
 * @since
 * @date 2014-11-20 下午5:51:11
 */
interface ConstantsService {
    @GET(ServerUrlImpl.CONSTANTS)
    operator fun get(callback: Callback<Result<ConstantsData?>?>?)

    /**
     * API可用的入口
     *
     * @return
     */
    @GET(ServerUrlImpl.CONSTANTS_API_ENDPOINT)
    fun apiEndpoing(): Result<ApiEndpointData?>?
}